docs/getting-started/walkthrough.md

```diff
@@ -24,177 +24,411 @@ specific language governing permissions and limitations
+          meta:
+            kind: http
+            type: example/greeter
+          spec:
+            endpoint: http://python-worker:8000/statefun
+            states:
+              - seen_count
+            maxNumBatchRequests: 500
+            timeout: 2min
+    ingresses:
+      - ingress:
+          meta:
+            type: statefun.kafka.io/routable-protobuf-ingress
+            id: example/names
+          spec:
+            address: kafka-broker:9092
+            consumerGroupId: my-group-id
+            topics:
+              - topic: names
+                typeUrl: com.googleapis/example.GreetRequest
+                targets:
+                  - example/greeter
+    egresses:
+      - egress:
+          meta:
+            type: statefun.kafka.io/generic-egress
+            id: example/greets
+          spec:
+            address: kafka-broker:9092
+            deliverySemantic:
+              type: exactly-once
+              transactionTimeoutMillis: 100000
+{% endhighlight %}
 
-You can check the full code for the application described in this walkthrough [here]({{ site.github_url }}/tree/{{ site.github_branch }}/statefun-examples/statefun-greeter-example).
-In particular, take a look at the module ``GreetingModule``, which is the main entry point for the full application, to see how everything gets tied together.
-You can run this example locally using the provided Docker setup.
+This configuration does a few interesting things.
 
-{% highlight bash %}
-$ docker-compose build 
-$ docker-compose up
+The first is to declare our function, `example/greeter`.
+It includes the endpoint by which it is reachable along with the states the function has access to.
+
+The ingress is the input Kafka topic that routes `GreetRequest` messages to the function.
+Along with basic properties like broker address and consumer group, it contains a list of targets.
+These are the functions each message will be sent to.
+
+The egress is the output Kafka cluster.
+It contains broker specific configurations but allows each message to route to any topic.
+
+## Deployment
+
+Now that the greeter application has been built it is time to deploy. 
+The simplest way to deploy a Stateful Function application is by using the community provided base image and loading your module.
+The base image provides the Stateful Function runtime, it will use the provided `module.yaml` to configure for this specific job.
+This can be found in the docker file in the root directory. 
+
+{% highlight docker %}
+FROM statefun
+
+RUN mkdir -p /opt/statefun/modules/greeter
+ADD module.yaml /opt/statefun/modules/greeter
 {% endhighlight %}
 
-Then, send some messages to the topic "names", and observe what comes out of "greetings".
+You can now run this application locally using the provided Docker setup.
 
 {% highlight bash %}
-$ docker-compose exec kafka-broker kafka-console-producer.sh \
-    --broker-list localhost:9092 \
-    --topic names
+$ docker-compose up -d
```

Comment:
Is this supposed to work?
```
Building master
Step 1/3 : FROM statefun
ERROR: Service 'master' failed to build: pull access denied for statefun, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

Reply:
Not yet because we haven't released :) In the mean time you can build the image locally using the `./build-distrubition.sh` script in the tools folder.

Reply:
Thx. That worked

```
Step 5/9 : RUN pip install -r requirements.txt
 ---> Running in 7a9ee7fda4f7
ERROR: Could not find a version that satisfies the requirement apache-flink-statefun (from -r requirements.txt (line 17)) (from versions: none)
ERROR: No matching distribution found for apache-flink-statefun (from -r requirements.txt (line 17))
ERROR: Service 'python-worker' failed to build: The command '/bin/sh -c pip install -r requirements.txt' returned a non-zero code: 1
```

How do I get `apache-flink-statefun` into that Docker build?

I managed to create a `RELEASE_VERSION=2.0 ./create_python_sdk_release.sh` release

Reply:
Update `greeter/Dockerfile` to be

```
FROM python:3.7-alpine

RUN mkdir -p /app
WORKDIR /app

COPY apache_flink_statefun-snapshot-py3-none-any.whl /app
RUN pip install apache_flink_statefun-snapshot-py3-none-any.whl

COPY requirements.txt /app
RUN pip install -r requirements.txt

COPY greeter.py /app
COPY messages_pb2.py /app

EXPOSE 8000

CMD ["gunicorn", "-b", "0.0.0.0:8000", "-w 4", "greeter:app"]
```

Again, just a stop gap until the first release. This is why I have the warning that we do not publish snapshot versions of the python sdk.